File: DotNet/Node.Core/API/ActionParameter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using DataFlow.Component.Interface;

namespace Node.Core.API
{
    /// <summary>
    /// ActionParameter is defined as basic parameter type to commnincate between the IActionComponent.
    /// </summary>
    public class ActionParameter : IActionParameter
    {
        //***********************************************************************
        // Public Members
        //***********************************************************************
        #region Public Members

        #endregion

        //***********************************************************************
        // Private Members
        //***********************************************************************
        #region Private Members
        private string name;
        private object value;
        private string type;
        private ActionParameterDirection direction = ActionParameterDirection.None;

        #endregion

        //***********************************************************************
        // Constructors
        //***********************************************************************
        #region Constructors

        #endregion

        //***********************************************************************
        // Delegate Events
        //***********************************************************************
        #region Delegate Events

        #endregion

        //***********************************************************************
        // Public Properties
        //***********************************************************************
        #region Public Properties
        /// <summary>
        /// Name of ActionParameter.
        /// </summary>
        public string ParameterName
        {
            get { return this.name; }
            set { this.name = value; }
        }
        /// <summary>
        /// Value of ActionParameter.
        /// </summary>
        public object ParameterValue
        {
            get { return this.value; }
            set { this.value = value; }
        }
        /// <summary>
        /// Type of ActionParameter.
        /// </summary>
        public string ParameterType
        {
            get { return this.type; }
            set { this.type = value; }
        }
        /// <summary>
        /// Direction of ActionParameter.
        /// </summary>
        public ActionParameterDirection Direction
        {
            get { return this.direction; }
            set { this.direction = value; }
        }

        /// <summary>
        /// Gets or sets the DEDL Encoding value.
        /// </summary>
        public string DEDLEncoding { get; set; }

        /// <summary>
        /// Gets or sets the DEDL Occurence Number value.
        /// </summary>
        public string DEDLOccurenceNumber { get; set; }

        /// <summary>
        /// Gets or sets the DEDL type value.
        /// </summary>
        public string DEDLType { get; set; }

        /// <summary>
        /// Gets or sets the DEDL RequiredIndicator value.
        /// </summary>
        public string DEDLRequiredIndicator { get; set; }
        /// <summary>
        /// Gets or sets the DEDL RequiredIndicator value.
        /// </summary>
        public string DEDLTypeDescriptor { get; set; }


        #endregion

        //***********************************************************************
        // Protected Properties
        //***********************************************************************
        #region Protected Properties

        #endregion

        //***********************************************************************
        // Public Methods
        //***********************************************************************
        #region Public Methods

        #endregion

        //***********************************************************************
        // Protected Methods
        //***********************************************************************
        #region Protected Methods

        #endregion

        //***********************************************************************
        // Private Methods
        //***********************************************************************
        #region Private Methods

        #endregion

        //***********************************************************************
        // Internal Handlers
        //***********************************************************************
        #region Internal Handlers

        #endregion
    }
}

File: EN Node for Java environment/Node/src/Node/API/ActionParameter.java
package Node.API;

import DataFlow.Component.Interface.*;
/**
 * <p>Implement ActionPrarameter</p>
 * <p>Company: enfoTech & Consulting, Inc.</p>
 * 
 * @author enfoTech
 * @version 2.0
 */

public class ActionParameter implements IActionParameter {
    private String name;
    private Object value;
    private String type;
    private int direction = ActionParameterDirection.Input;
    
    /**
     * Gets the parameter direction value. The values of direction are defined in ActionParameterDirection class. 
     */
	public int getDirection() {
		return direction;
	}
	
	/**
	 * Sets the parameter direction. The values of direction are defined in ActionParameterDirection class.
	 */
	public void setDirection(int direction) {
		this.direction = direction;
	}
	
	/**
	 * Gets the parameter name.
	 */
	public String getParameterName() {
		return name;
	}
	
	/**
	 * Sets the parameter name.
	 */
	public void setParameterName(String name) {
		this.name = name;
	}
	
	/**
	 * Gets the parameter type.
	 */
	public String getParameterType() {
		return type;
	}
	
	/**
	 * Sets the parameter type.
	 */
	public void setParameterType(String type) {
		this.type = type;
	}
	
	/**
	 * Gets the parameter value.
	 */
	public Object getParameterValue() {
		return value;
	}
	
	/**
	 * Sets the parameter value.
	 */
	public void setParameterValue(Object value) {
		this.value = value;
	}
    
    
}
